secboot/secboot_tpm.go

```diff
@@ -433,30 +500,33 @@ func ResealKeys(params *ResealKeysParams) error {
 		return fmt.Errorf("cannot read the policy auth key file %s: %w", params.TPMPolicyAuthKeyFile, err)
 	}
 
-	hasOldObject := false
-	hasNewData := false
-
 	keyDatas := make([]*sb.KeyData, 0, numSealedKeyObjects)
 	sealedKeyObjects := make([]*sb_tpm2.SealedKeyObject, 0, numSealedKeyObjects)
 	for _, keyfile := range params.KeyFiles {
-		keyData, keyObject, err := readKeyFile(keyfile)
+		loadedKey := &defaultKeyLoader{}
+		const hintExpectFDEHook = false
+		err := readKeyFile(keyfile, loadedKey, hintExpectFDEHook)
 		if err != nil {
 			return fmt.Errorf("cannot read key file %s: %w", keyfile, err)
 		}
-		keyDatas = append(keyDatas, keyData)
-		sealedKeyObjects = append(sealedKeyObjects, keyObject)
-		if keyObject == nil {
-			hasNewData = true
-		} else {
-			hasOldObject = true
+		if loadedKey.SealedKeyObject != nil {
```

Comment:
we should maybe leave a TODO about refactor some of these  ifs into methods on the loadedKey interface, but also I as wrote elsewhere wondering if there will be other implementations?

Reply:
I have added a TODO to the definition of `keyLoader` interface.